### PR TITLE
Add Event Grid Blob trigger

### PR DESCRIPTION
**Pull request by castrodd**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [X] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs [hunk 1]
```diff
@@ -86,6 +86,7 @@ public static async Task DeployDotnetFunction(string templateName, string functi
         private static string GetTemplateShortName(string templateName) => templateName.ToLowerInvariant() switch
         {
             "blobtrigger" => "blob",
+            "eventgridblobtrigger" => "egblob",
```
Comment (ejizba) [inline src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs:89]: I think we want this to match the short name here: https://github.com/Azure/azure-functions-templates/pull/1519/files#diff-f46c6528d07abea982faaa52631c315020e217f2e08caaada8997b3e53cbd237R12
```suggestion
            "eventgridblobtrigger" => "eventgridblob",
```

Comment (castrodd) [reply src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs]: Updated

File: src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs [hunk 2]
```diff
@@ -86,6 +86,7 @@ public static async Task DeployDotnetFunction(string templateName, string functi
         private static string GetTemplateShortName(string templateName) => templateName.ToLowerInvariant() switch
```
Comment (ejizba) [inline src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs:86]: Tested locally and looks good to me. We'll just need to update the nupkg versions once those are out. Example here: https://github.com/Azure/azure-functions-core-tools/pull/3493

Comment (castrodd) [reply src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs:86]: Updated to 4.0.3038

**Review by khkh-ms — APPROVED** [review]

File: build/Settings.cs [hunk 3]
```diff
@@ -19,11 +19,11 @@ private static string config(string @default = null, [CallerMemberName] string k
...
-        public const string DotnetIsolatedItemTemplatesVersion = "4.0.2945";
-        public const string DotnetIsolatedProjectTemplatesVersion = "4.0.2945";
-        public const string DotnetItemTemplatesVersion = "4.0.2945";
-        public const string DotnetProjectTemplatesVersion = "4.0.2945";
-        public const string TemplateJsonVersion = "3.1.1648";
+        public const string DotnetIsolatedItemTemplatesVersion = "4.0.3038";
+        public const string DotnetIsolatedProjectTemplatesVersion = "4.0.3038";
+        public const string DotnetItemTemplatesVersion = "4.0.3038";
+        public const string DotnetProjectTemplatesVersion = "4.0.3038";
+        public const string TemplateJsonVersion = "4.0.3038";
```
Comment (khkh-ms) [inline build/Settings.cs:26]: You don't have to update the TemplateJsonVersion.

**Review by khkh-ms — CHANGES_REQUESTED** [review]

File: build/Settings.cs (hunk 3 above)
Comment (castrodd) [reply build/Settings.cs]: Fixed

File: build/Settings.cs [hunk 4]
```diff
@@ -19,11 +19,11 @@ private static string config(string @default = null, [CallerMemberName] string k
...
-        public const string DotnetIsolatedItemTemplatesVersion = "4.0.2945";
-        public const string DotnetIsolatedProjectTemplatesVersion = "4.0.2945";
-        public const string DotnetItemTemplatesVersion = "4.0.2945";
-        public const string DotnetProjectTemplatesVersion = "4.0.2945";
-        public const string TemplateJsonVersion = "3.1.1648";
+        public const string DotnetIsolatedItemTemplatesVersion = "4.0.3038";
+        public const string DotnetIsolatedProjectTemplatesVersion = "4.0.3038";
+        public const string DotnetItemTemplatesVersion = "4.0.3038";
+        public const string DotnetProjectTemplatesVersion = "4.0.3038";
+        public const string TemplateJsonVersion = "3.1.1648;
```
Comment (khkh-ms) [inline build/Settings.cs:26]: It is missing the ending quote.

Comment (castrodd) [reply build/Settings.cs]: Sorry, fixed typo.

**Review by khkh-ms — APPROVED** [review]

**Pull request merged**